packages/driver/src/cypress/runner.ts

```diff
@@ -505,26 +502,18 @@ const normalizeAll = (suite, initialTests = {}, setTestsById, setTests, onLogsBy
   // create optimized lookups for the tests without
   // traversing through it multiple times
   const tests: Record<string, any> = {}
-  const normalizedSuite = normalize(suite, tests, initialTests, onLogsById, getRunnableId, getHookId, getOnlyTestId, getOnlySuiteId, createEmptyOnlyTest)
-
-  if (setTestsById) {
-    // use callback here to hand back
-    // the optimized tests
-    setTestsById(tests)
-  }
+  const normalizedSuite = normalize(suite, tests, previouslyRanTests, onLogsById, getRunnableId, getHookId, getOnlyTestId, getOnlySuiteId, createEmptyOnlyTest)
 
-  if (setTests) {
-    let i = 0
+  let i = 0
 
-    const testsArr = _.map(tests, (test) => {
-      test.order = i += 1
+  const testsArr = _.map(tests, (test) => {
+    test.order = i += 1
```

Comment:
is `tests` an array in this case? I think the second argument of the callback is the index and likely gets rid of `i`.
```suggestion
  const testsArr = _.map(tests, (test, index) => {
    test.order = index + 1
```